### PR TITLE
test: added support for uielement

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml [hunk 1]
```diff
@@ -13,27 +13,22 @@
 	<Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
 
 		<Grid.RowDefinitions>
-			<RowDefinition x:Name="TopPaddingRow"
-						   Height="0" />
-			<RowDefinition Height="Auto" />
+			<RowDefinition x:Name="TopPaddingRow" Height="0" />
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml:16]: to remove: navbar bar fix got slipped in...

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs [hunk 2]
```diff
@@ -221,5 +215,13 @@ private void NavigationViewControl_SizeChanged(object sender, SizeChangedEventAr
 				NavigationViewControl.PaneDisplayMode = MUXC.NavigationViewPaneDisplayMode.LeftMinimal;
 			}
 		}
+
+		private void NavigationViewControl_DisplayModeChanged(MUXC.NavigationView sender, MUXC.NavigationViewDisplayModeChangedEventArgs e)
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs:219]: to remove: navbar bar fix got slipped in...

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs [hunk 3]
```diff
@@ -77,8 +77,7 @@ private void SetDarkLightToggleInitialState()
 		/// </summary>
 		private void InitializeSafeArea()
 		{
-#if !IS_WINUI
-			var full = XamlWindow.Current.Bounds;
+			var full = App.Instance.Window.Bounds;
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs:80]: this should be kept however

File: src/Directory.Packages.props [hunk 4]
```diff
@@ -20,6 +20,7 @@
     <PackageVersion Include="Newtonsoft.Json" Version="13.0.1" />
     <PackageVersion Include="NUnit" Version="3.13.3" />
     <PackageVersion Include="NUnit3TestAdapter" Version="4.3.0-alpha-net7.4" />
+    <PackageVersion Include="Uno.UI.RuntimeTests.Engine" Version="0.1.0-dev.32" />
```
Comment (Xiaoy312) [inline src/Directory.Packages.props:23]: waiting on unoplatform/uno.ui.runtimetests.engine#59
and need to update package

**Comment by Xiaoy312** [comment]
todo:
- [ ] bump package after rte#59
- [ ] drop EngineMetaAttributes.cs and drop `UNO_RUNTIMETESTS_DISABLE_*` constants

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs (hunk 3 above)
Comment (Xiaoy312) [reply samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs]: should not be... ApplicationView is what crashes winui-desktop

File: src/Uno.Toolkit.RuntimeTests/Tests/LeakTest.cs [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+			};
+
+			var _holders = new ConditionalWeakTable<DependencyObject, Holder>();
+			void TrackDependencyObject(DependencyObject target) => _holders.Add(target, new Holder(HolderUpdate));
+
+			var maxCounter = 0;
+			var activeControls = 0;
+			var maxActiveControls = 0;
+
+			var rootContainer = new ContentControl();
```
Comment (kazo0) [inline src/Uno.Toolkit.RuntimeTests/Tests/LeakTest.cs:76]: Just comparing this to https://github.com/unoplatform/uno/blob/master/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs

They have a `_holder.Reset()` before setting the `rootContainer`, could it maybe a reason they're all failing?

Comment (Xiaoy312) [reply src/Uno.Toolkit.RuntimeTests/Tests/LeakTest.cs:76]: that was added recently; but doesnt help since even the first test is already failing...

**Review by kazo0 — APPROVED** [review]

**Pull request merged**